packages/dev/core/src/Engines/WebGL/webGLRenderTargetWrapper.ts

```diff
@@ -75,23 +75,46 @@ export class WebGLRenderTargetWrapper extends RenderTargetWrapper {
      * Binds a texture to this render target on a specific attachment
      * @param texture The texture to bind to the framebuffer
      * @param attachmentIndex Index of the attachment
-     * @param faceIndex The face of the texture to render to in case of cube texture
+     * @param faceIndexOrLayer The face or layer of the texture to render to in case of cube texture or array texture
      * @param lodLevel defines the lod level to bind to the frame buffer
      */
-    private _bindTextureRenderTarget(texture: InternalTexture, attachmentIndex: number = 0, faceIndex: number = -1, lodLevel: number = 0) {
+    private _bindTextureRenderTarget(texture: InternalTexture, attachmentIndex: number = 0, faceIndexOrLayer?: number, lodLevel: number = 0) {
         if (!texture._hardwareTexture) {
             return;
         }
 
-        const gl = this._context;
         const framebuffer = this._framebuffer;
 
         const currentFB = this._engine._currentFramebuffer;
         this._engine._bindUnboundFramebuffer(framebuffer);
-        const attachment = (<any>gl)[this._engine.webGLVersion > 1 ? "COLOR_ATTACHMENT" + attachmentIndex : "COLOR_ATTACHMENT" + attachmentIndex + "_WEBGL"];
-        const target = faceIndex !== -1 ? gl.TEXTURE_CUBE_MAP_POSITIVE_X + faceIndex : gl.TEXTURE_2D;
 
-        gl.framebufferTexture2D(gl.FRAMEBUFFER, attachment, target, texture._hardwareTexture.underlyingResource, lodLevel);
+        if (this._engine.webGLVersion > 1) {
+            const gl = this._context as WebGL2RenderingContext;
+
+            const attachment = (<any>gl)["COLOR_ATTACHMENT" + attachmentIndex];

```

Comment:
I wonder if splitting this function per type (2dArray/3d, cube, others) would not make sense it becomes quite complex what happens inside also this is only called in setFace... which already checks the different cases.

Reply:
I'm worried about calls to `setTexture`, although I agree in every other case it is redundant